typescript/agentkit/src/wallet-providers/privyEvmDelegatedEmbeddedWalletProvider.ts

```diff
@@ -0,0 +1,500 @@
+
+      return receipt.transactionHash;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Native transfer failed: ${error.message}`);
+      }
+      throw new Error("Native transfer failed");
+    }
+  }
+
+  /**
+   * Exports the wallet information.
+   *
+   * @returns The wallet data
+   */
+  exportWallet(): PrivyWalletExport {
+    return {
+      walletId: this.#walletId,
+      authorizationPrivateKey: this.#authKey,
+      networkId: this.#network.networkId!,
+      chainId: this.#network.chainId,
+    };
+  }
+
+  /**
+   * Generate Privy authorization signature for API requests
+   *
+   * @param url - The URL for the request
+   * @param body - The request body
+   * @returns The generated signature
+   */
+  private generatePrivySignature(url: string, body: object): string {
+    try {
+      const payload = {
+        version: 1,
+        method: "POST",
+        url,
+        body,
+        headers: {
+          "privy-app-id": this.#appId,
+        },
+      };
+
+      const serializedPayload = canonicalize(payload);
+      if (!serializedPayload) throw new Error("Failed to canonicalize payload");
+
+      const serializedPayloadBuffer = Buffer.from(serializedPayload);
+
+      const privateKeyAsString = this.#authKey.replace("wallet-auth:", "");
+      const privateKeyAsPem = `-----BEGIN PRIVATE KEY-----\n${privateKeyAsString}\n-----END PRIVATE KEY-----`;
+
+      const privateKey = crypto.createPrivateKey({
+        key: privateKeyAsPem,
+        format: "pem",
+      });
+
+      const signatureBuffer = crypto.sign("sha256", serializedPayloadBuffer, privateKey);
+      return signatureBuffer.toString("base64");
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Error generating Privy authorization signature: ${error.message}`);
+      }
+      throw new Error("Error generating Privy authorization signature");
+    }
+  }
+
+  /**
+   * Get Privy headers for API requests
+   *
+   * @param url - The URL for the request
+   * @param body - The request body
+   * @returns The headers for the request
+   */
+  private getPrivyHeaders(url: string, body: object) {
+    return {
+      "Content-Type": "application/json",
+      Authorization: `Basic ${Buffer.from(`${this.#appId}:${this.#appSecret}`).toString("base64")}`,
+      "privy-app-id": this.#appId,
+      "privy-authorization-signature": this.generatePrivySignature(url, body),
+    };
+  }
+
+  /**
+   * Execute a Privy API request.
+   *
+   * @param body - The request body to send to the Privy API
+   * @returns A promise that resolves to the response data
+   * @throws Error if the request fails
+   */
+  private async executePrivyRequest<T>(body: Record<string, unknown>): Promise<T> {
+    const url = `https://api.privy.io/v1/wallets/rpc`;
+    const headers = this.getPrivyHeaders(url, body);
+
+    try {
+      const response = await axios.post<T>(url, body, { headers });
+      return response.data;
```

Comment:
not blocking, but i would have liked to use native `fetch` instead of axios, since we use `fetch` already in a few places, and we're not doing anything special with axios here